### ISSUE TITLE
`cbindgen` temporarily requires nightly

### ISSUE THREAD

**Issue by eqrion** [eqrion/cbindgen#147]
TL;DR `cbindgen` temporarily requires nightly because of dependencies and fixes are being investigated.

Due to updating to syn `0.12` which brought in a dependency on an internal compiler lib (`rustc/libproc-macro`), `cbindgen` stopped working outside of `cargo run` with:
```
dyld: Library not loaded: @rpath/libproc_macro-89628dcfdc0f6129.dylib
  Referenced from: /Users/rhunt/.cargo/bin/cbindgen
  Reason: image not found
fish: 'cbindgen' terminated by signal SIGABRT (Abort)
```

`syn:0.12` was a big update that required many changes to cbindgen and added much better parsing and error messages, so downgrading back to `0.11` is not desirable.

To fix this, we forked `syn` and published snapshots of `rustc/libproc-macro` and used some other published snapshots of dependencies of `rustc/libproc-macro`. This works, but unfortunately requires nightly.

Since then `proc-macro2` has been updated to have a feature to disable linking with `rustc/libproc-macro` which should fix this issue. `syn` and other users(?) of `proc-macro2` will need to update for this to work. Right now `syn` is waiting for enough changes to justify a breaking change before doing this.

In the meantime we updated our fork of syn to use this flag and published `0.5.1` with it. Unfortunately for some reason this approach worked locally, but when run through `cargo install` the linking error is back. So that version has been yanked.

This is the current status of the issue, I'll post updates here. Apologies for any inconveniences this has caused.

**Comment by Luke-Nukem** [comment]
Darn. I was wondering why things were suddenly breaking in my stable builds.
What was the last version to be buildable on stable?

**Comment by eqrion** [comment]
@Luke-Nukem I believe that 0.4.3 is the last one buildable on stable.

**Comment by eqrion** [comment]
This has been resolved in 0.5.2.
